src/gas_metering/mod.rs

```diff
@@ -468,6 +501,38 @@ fn add_grow_counter<R: Rules>(
 	b.build()
 }
 
+fn add_gas_counter(module: elements::Module, gas_global: u32) -> elements::Module {
+	use parity_wasm::elements::Instruction::*;
+
+	let mut b = builder::from_module(module);
+	b.push_function(
```

Comment:
Add a comment explaining why we're using a function.

Reply:
Added/updated the top `inject` comment